libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
+        yield();
```

Comment:
I mean GPIO interrupt and SPI transaction should happen inside the interrupt handler.
I want to read the data from MCP3914 immediately when it is ready (attaching GPIO interrupt to the DR assigned GPIO pin) and probably several times during a single `loop`, even during WiFi, TCP/IP or other tasks (if/when possible) that may take too long to complete.
Other sensors I will be using, do allow the use of a flag, but not this one, so i will probably just disable the interrupt while working with SPI inside `loop` and re-enable it again.